website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -675,10 +746,31 @@ the types of the input and output elements. If your `DoFn` processes incoming
 look like this:
 {{< /paragraph >}}
 
+{{< paragraph class="language-go" >}}
+A `DoFn` processes one element at a time from the input `PCollection`. When you
+create a `DoFn` struct, you'll need to provide type parameters that match
+the types of the input and output elements in a ProcessElement method.
+If your `DoFn` processes incoming `string` elements and produces `int` elements
+for the output collection (like our previous example, `ComputeWordLengthFn`), your dofn could
+look like this:
+{{< /paragraph >}}
+
 {{< highlight java >}}
 static class ComputeWordLengthFn extends DoFn<String, Integer> { ... }
 {{< /highlight >}}
 
+{{< highlight go >}}
+type ComputeWordLengthFn struct{}
```

Comment:
Suggested change:

```
type ComputeWordLengthFn struct{}
```

to

```
// ComputeWordLengthFn is a DoFn that computes the word length of string elements.
type ComputeWordLengthFn struct{}
```

Reply:
I was 50/50 on including full Go comment in some of the examples. Since this is first one it does make sense in some way to include them. I worry that they could be a distracting detail though.